File: app/src/main/java/com/example/artur/watch/Model/Usuario.java
package com.example.artur.watch.Model;

import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;

@Entity
public class Usuario {

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @Id private long id;
    private String nome;
    private String username;
    private String email;
    private String senha;

    public Usuario(){}

    public Usuario(String nome, String username, String email, String senha) {
        this.nome = nome;
        this.username = username;
        this.email = email;
        this.senha = senha;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }
}

File: app/src/main/java/com/example/artur/watch/Model/Usuario.kt
package com.example.artur.watch.Model

import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import io.objectbox.annotation.Unique

@Entity
class Usuario{

    @Id
    var id: Long = 0
    lateinit var nome: String
    @Unique lateinit var username: String
    lateinit var email: String
    lateinit var senha: String

    constructor()

    constructor(nome: String, username: String, email: String, senha: String) {
        this.nome = nome
        this.username = username
        this.email = email
        this.senha = senha
    }

    override fun toString(): String {
        return "Nome: ${this.nome}\n" +
                "Nome de usuário: ${this.username}\n" +
                "E-mail: ${this.email}\n"
    }
}
